Library/Homebrew/cmd/migrate.rb

```diff
@@ -0,0 +1,29 @@
+require "migrator"
+require "formula_renames"
+
+module Homebrew
+  def migrate
+    raise FormulaUnspecifiedError if ARGV.named.empty?
+
+    ARGV.named.each do |name|
```

Comment:
Because if we install core `oldname` and then it goes to `newname`, but `oldname` from tap stil exists, the wrong formula will be instantiated, because we instantiate formulae this way:

1. name
2. tap name
3. oldname
4. tap oldname